### ISSUE TITLE
Pass --exclude-config programatically

### ISSUE THREAD

**Issue by ppalaga** [quarkusio/quarkus#17482]
`--exclude-config` was introduced in https://github.com/oracle/graal/pull/3179
It would be nice to be able to pass it programatically from a Quarkus extension

**Comment by geoand** [comment]
@galderz do you plan on making this available for easy use in Quarkus?

**Comment by galderz** [comment]
@geoand No, that's not something I'm looking into.

**Comment by geoand** [comment]
Thanks for the info @galderz.

I assume that at some point this is something we would want to have in Quarkus

**Comment by geoand** [comment]
@Sanne this is something you wanted as well, right?

**Comment by Sanne** [comment]
> @Sanne this is something you wanted as well, right?

Right! We have an immediate need in the case of the Oracle JDBC driver, and I totally expect more libraries in the future to occasionally include some mistakes, which we would only be able to work around when we have this.

**Comment by geoand** [comment]
@galderz how would one access this programmatically from a Feature?

**Comment by galderz** [comment]
@geoand I don't think doing this a Feature is possible. The code that excludes jars/resources works in the 1st stage of native-image, the driver phase. This phase constructs a java invocation for the second phase and hence we exclude things to being passed on to the 2nd phase, where Feature instances and others come into play.

I would expect that each extension would define 0..N `exclude-config` parameters (see examples [here](https://github.com/oracle/graal/pull/3179)), and then native image would take all of those and pass them in as parameters to `native-image` invocation.

**Comment by geoand** [comment]
Gotcha. I'll look into it hopefully sometime soon

**Comment by Sanne** [comment]
Features should be able to change the flags which we feed into the `native-image` process

**Comment by galderz** [comment]
Ah, you mean Quarkus Features? I thought @geoand was referring to GraalVM Features... :)

**Comment by geoand** [comment]
Yes, I did mean GraalVM Features

**Comment by Sanne** [comment]
ah ok. Yes it definitely needs to be at  the level of command line arguments, anything else would be too late.

**Comment by geoand** [comment]
I opened https://github.com/quarkusio/quarkus/pull/19649, but the problem is that I can't seem to get `native-image` work with multiple `--exclude-config` options, so I'll need @galderz's help with that
